Building OpenThread NCP: build system has concurrency issue

**Describe the bug**

Rarely I do get build errors when building samples/net/openthread/ncp and other openthread samples that also require errno_private.h:

```
In file included from /home/markus/src/zephyrproject/zephyr/lib/libc/minimal/include/errno.h:20,
                 from /home/markus/src/zephyrproject/modules/lib/openthread/src/lib/spinel/spinel.c:48:
/home/markus/src/zephyrproject/zephyr/include/sys/errno_private.h:34:10: fatal error: syscalls/errno_private.h: No such file or directory
   34 | #include <syscalls/errno_private.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

Seems like a concurrency issue in the build system, where the generated file `./build/zephyr/include/generated/syscalls/errno_private.h` is not created by the time spinel.c is being built.

**To Reproduce**
See above.

**Expected behavior**
No concurrency issue.

**Impact**
annoyance. 

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA 04d3c3c1a0d479aada6bab83b3d29e6bf61a2a6a

> Ok, I've reproduced it, but it appears only to happen with changes from #28458, when RCP variant is built. To reproduce:
> `west build -b nrf52840dk_nrf52840 -p -- -DCONF_FILE=prj-rcp.conf`

yep, happens on #28458:

It is enough to run:
```
$ cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -Bbuild  samples/net/openthread/ncp/
$ ninja -Cbuild openthread-spinel-rcp -v
...
FAILED: modules/openthread/build/src/lib/spinel/CMakeFiles/openthread-spinel-rcp.dir/spinel.c.obj 
/opt/gcc-arm-none-eabi-7-2018-q2-update/bin/arm-none-eabi-gcc -DMBEDTLS_CONFIG_FILE=\"config-tls-generic.h\" -DOPENTHREAD_CONFIG_BORDER_AGENT_ENABLE=1 -DOPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE=1 -DOPENTHREAD_CONFIG_COMMISSIONER_ENABLE=1 -DOPENTHREAD_CONFIG_DHCP6_SERVER_ENABLE=1 -DOPENTHREAD_CONFIG_ENABLE_BUILTIN_MBEDTLS=0 -DOPENTHREAD_CONFIG_ENABLE_BUILTIN_MBEDTLS_MANAGEMENT=0 -DOPENTHREAD_CONFIG_LOG_LEVEL=0 -DOPENTHREAD_CONFIG_NCP_UART_ENABLE=1 -DOPENTHREAD_CONFIG_THREAD_VERSION=OT_THREAD_VERSION_1_1 -DOPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE=1 -DOPENTHREAD_CONFIG_UDP_FORWARD_ENABLE=1 -DOPENTHREAD_ENABLE_NCP=1 -DOPENTHREAD_PROJECT_CORE_CONFIG_FILE=\"openthread-core-zephyr-config.h\" -DOPENTHREAD_RADIO=1 -DOPENTHREAD_SPINEL_CONFIG_OPENTHREAD_MESSAGE_ENABLE=0 -DPACKAGE_NAME=\"OPENTHREAD\" -DPACKAGE_VERSION=\"g2ac83762b\" -Imodules/openthread/build/etc/cmake -I/projects/github/ncs/modules/lib/openthread/etc/cmake -I/projects/github/ncs/modules/lib/openthread/include -I/subsys/net/lib/openthread/platform -I/projects/github/ncs/modules/crypto/mbedtls/configs -I/projects/github/ncs/modules/crypto/mbedtls/include -I/projects/github/ncs/modules/lib/openthread/examples/platforms/zephyr -I/projects/github/ncs/modules/lib/openthread/src -I/projects/github/ncs/modules/lib/openthread/src/core -Os -imacros/projects/github/ncs/zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -march=armv7e-m -imacros/projects/github/ncs/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -ffunction-sections -fdata-sections -fno-builtin -I/projects/github/ncs/zephyr/include -I/projects/github/ncs/zephyr/build/zephyr/include/generated -I/projects/github/ncs/zephyr/soc/arm/nordic_nrf/nrf52 -I/projects/github/ncs/zephyr/lib/libc/minimal/include -I/projects/github/ncs/zephyr/subsys/settings/include -I/projects/github/ncs/modules/hal/cmsis/CMSIS/Core/Include -I/projects/github/ncs/modules/hal/nordic/drivers/nrf_radio_802154/. -I/projects/github/ncs/modules/hal/nordic/nrfx -I/projects/github/ncs/modules/hal/nordic/nrfx/drivers/include -I/projects/github/ncs/modules/hal/nordic/nrfx/mdk -I/projects/github/ncs/modules/hal/nordic/. -isystem/projects/github/ncs/zephyr/lib/libc/minimal/include   -Wall -Wextra -Wshadow -Werror -pedantic-errors -MD -MT modules/openthread/build/src/lib/spinel/CMakeFiles/openthread-spinel-rcp.dir/spinel.c.obj -MF modules/openthread/build/src/lib/spinel/CMakeFiles/openthread-spinel-rcp.dir/spinel.c.obj.d -o modules/openthread/build/src/lib/spinel/CMakeFiles/openthread-spinel-rcp.dir/spinel.c.obj   -c /projects/github/ncs/modules/lib/openthread/src/lib/spinel/spinel.c
In file included from /projects/github/ncs/zephyr/lib/libc/minimal/include/errno.h:20:0,
                 from /projects/github/ncs/modules/lib/openthread/src/lib/spinel/spinel.c:48:
/projects/github/ncs/zephyr/include/sys/errno_private.h:34:10: fatal error: syscalls/errno_private.h: No such file or directory
 #include <syscalls/errno_private.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
so the dependency issue is introduced in that PR.

--- a/west.yml
+++ b/west.yml
@@ -111,7 +111,7 @@ manifest:
       revision: 3f545d76a2e6d1db83a470ccdb5bebd1f484e137
       path: modules/lib/loramac-node
     - name: openthread
-      revision: 7854c687bf23ce509a65d118cbea0c7011a89a1e
+      revision: 07f430dac6c400abdd358ba1173c0f94b94a5a80
       path: modules/lib/openthread
     - name: segger
       revision: 874d9e9696b00c09f9eeefe839028dc25fe44983

